File: C/Algorithms/Sorting/SelectionSort.c
#include<stdio.h>
int n;

void printArray(int arr[]){
	printf("Displaying the array\n");
	for(int i=0;i<n;i++){
		printf("%d ",arr[i]);
	}
	printf("\n");
}

void swap(int *a, int *b){
	int temp = *a;
	*a = *b;
	*b = temp;
}

int getMaxIndex(int arr[], int end){
	int max = 0;
	for(int i = 0;i<=end;i++){
		if(arr[max] < arr[i]){
			max = i;
		}
	}
	return max;
}

void selectionSort(int arr[]){
	for(int i=0;i<n-1;i++){
		int lastIndex = n - i - 1;
		int maxElementIndex = getMaxIndex(arr,lastIndex);
		swap(&arr[maxElementIndex], &arr[lastIndex]);
	}
}

void main(){
	printf("Enter the number of elements\n");
	scanf("%d",&n);
	int arr[n];
	printf("Enter the array elements\n");
	for(int i=0;i<n;i++){
		scanf("%d",&arr[i]);
	}
	selectionSort(arr);
	printArray(arr);
}

File: Python/Practice/SelectionSort.py
def getmax(list, e):
    max = 0
    for i in range(e+1):
        if(list[i] > list[max]):
            max = i
            
    return max

def selectionSort(list):
    for i in range(len(list)):
        maxIndex = getmax(list,len(list)-i-1)
        list[len(list) - i - 1], list[maxIndex] =  list[maxIndex], list[len(list) - i - 1]

# list = [2,1,4,6,7,5,3]
list = [23,12,45,67,34,89,57,90]
selectionSort(list)
print(list)